File: Roman to Integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        int answer = 0;
        
        for(int i = 0; i < s.size(); i++){
            if(s[i] == 'I'){
                if(i + 1 < s.size()){
                    if(s[i + 1] == 'V'){
                        answer += 4;
                        i++;
                    }
                    else if(s[i + 1] == 'X'){
                        answer += 9;
                        i++;
                    }
                    else{
                        answer += 1;
                    }
                }
                else{
                    answer += 1;
                }
            }
            else if(s[i] == 'V'){
                answer += 5;
            }
            else if(s[i] == 'X'){
                if(i + 1 < s.size()){
                    if(s[i + 1] == 'L'){
                        answer += 40;
                        i++;
                    }
                    else if(s[i + 1] == 'C'){
                        answer += 90;
                        i++;
                    }
                    else{
                        answer += 10;
                    }
                }
                else{
                    answer += 10;
                }
            }
            else if(s[i] == 'L'){
                answer += 50;
            }
            else if(s[i] == 'C'){
                if(i + 1 < s.size()){
                    if(s[i + 1] == 'D'){
                        answer += 400;
                        i++;
                    }
                    else if(s[i + 1] == 'M'){
                        answer += 900;
                        i++;
                    }
                    else{
                        answer += 100;
                    }
                }
                else{
                    answer += 100;
                }
            }
            else if(s[i] == 'D'){
                answer += 500;
            }
            else if(s[i] == 'M'){
                answer += 1000;
            }
        }
        
        return answer;
    }
};

File: Roman to Integer.js
/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function(s) {
    const map = new Map();
    const length = s.length;
    let answer = 0;
    
    map.set("I", 1);
    map.set("IV", 4);
    map.set("IX", 9);
    map.set("V", 5);
    map.set("X", 10);
    map.set("XL", 40);
    map.set("XC", 90);
    map.set("L", 50);
    map.set("C", 100);
    map.set("CD", 400);
    map.set("CM", 900);
    map.set("D", 500);
    map.set("M", 1000);
    
    for(let i = 0; i < s.length; i++){
        const s1 = map.get(s.slice(i, i + 1));
        const s2 = map.get(s.slice(i, i + 2));
        
        if(s2){
            answer += s2;
            i++;
        }
        else{
            answer += s1;
        }
    }
    
    return answer;
};